packages/appium/lib/main.js

```diff
@@ -138,6 +140,52 @@ function getExtraMethodMap(driverClasses, pluginClasses) {
     {}
   );
 }
+
+/**
+ * Prepares and validates appium home path folder
+ *
+ * @param {string} name The name of the appium home source (needed for error messages)
+ * @param {string} appiumHome The actual value to be verified
+ * @returns {Promise<string>} Same appiumHome value
+ * @throws {Error} If the validation has failed
+ */
+async function prepareAppiumHome(name, appiumHome) {
+  let stat;
+  try {
+    stat = await fs.stat(appiumHome);
+  } catch (e) {
+    let err = e;
+    if (e.code === 'ENOENT') {
+      try {
+        await fs.mkdir(appiumHome, {recursive: true});
+        return appiumHome;
+      } catch (e1) {
+        err = e1;
+      }
+    }
+    throw new Error(
+      `The path '${appiumHome}' provided in ${name} must be pointing ` +
+      `to a valid folder writeable for the current user account '${os.userInfo().username}'. ` +
+      `Original error: ${err.message}`
+    );
+  }
+  if (!stat.isDirectory()) {
+    throw new Error(
+      `The path '${appiumHome}' provided in ${name} must be pointing to a valid folder`
+    );
+  }
+  try {
+    await fs.access(appiumHome, fs.constants.W_OK);
+  } catch (e) {
+    logger.warn(
+      `The folder path '${appiumHome}' provided in ${name} is not ` +
+      `writeable for the current user account '${os.userInfo().username}'. This might ` +
```

Comment:
I'd probably want to be more firm in the error message; `APPIUM_HOME` _must_ be writable by the user running Appium.

Reply:
I had it like that initially, but it would probably prevent the server from running at all if it is somehow packed together with drivers. So I decided to be more flexible there

Reply:
I need to test locally first how recursive mkdir reacts to different exceptional situations

Reply:
I prefer to be permissive rather than protective. E.g. if they want to do something wacky then let them do it. @jlipps @KazuCocoa what is your opinion on that, shall I change the warning to an exception?

Reply:
I don't have a strong feeling either way. The important thing is that the message is informative, which it now is. The advantage of making it an error is that users will be sure to see the message and it won't get lost. I'm not sure of the advantage of being permissive about a case we're reasonably sure won't work (unless say a user is explicitly using only the npm based module detection instead of appium home and don't want any writeable folder, but we'd probably need to do more work to support that case anyway).

Reply:
ok, changed to an error